src/renderer/object/texture_region.rs

```diff
@@ -0,0 +1,36 @@
+#[derive(Clone, Copy, Debug, Default)]
+pub struct TextureRegion {
+    pub x: f32,
+    pub y: f32,
+    pub w: f32,
+    pub h: f32,
+}
+
+impl TextureRegion {
+    pub fn new() -> TextureRegion {
+        TextureRegion {
+            x: 0.0,
+            y: 0.0,
+            w: 1.0,
+            h: 1.0,
+        }
+    }
+
+    pub fn from_slice(subt: [f32; 4]) -> TextureRegion {
```

Comment:
Not sure this makes sense. Why not be explicit:
```
TextureRegion {x: 0.0, y: 0.0, w: 1.0, h: 1.0}
```

Reply:
Struct expressions are still kinda new to me and I forget :O